src/adapter/src/catalog/migrate.rs

```diff
@@ -89,6 +92,26 @@ pub(crate) async fn migrate(catalog: &mut Catalog) -> Result<(), anyhow::Error>
 // AST migrations -- Basic AST -> AST transformations
 // ****************************************************************************
 
+// Mark all current subsources as "references" subsources in anticipation of
+// adding "progress" subsources.
+fn subsource_type_option_rewrite(stmt: &mut mz_sql::ast::Statement<Raw>) {
+    if let Statement::CreateSubsource(CreateSubsourceStatement { with_options, .. }) = stmt {
+        if !with_options.iter().any(|option| {
+            matches!(
+                option.name,
+                CreateSubsourceOptionName::Progress | CreateSubsourceOptionName::References
+            )
+        }) {
+            with_options.push(mz_sql::ast::CreateSubsourceOption {
+                name: CreateSubsourceOptionName::References,
```

Comment:
Light desire to share a codepath with purification, but am not sure it's worth it. Thoughts?

Reply:
Given that we can now delete migrations, unifying the two seems like more work than it's worth. If you have a clean abstraction in mind, glad to do the typing, though.

Reply:
If we can delete it after one release cycle, can you also document the requirements for removing this migration?

Reply:
Oh sure--good call.